### ISSUE TITLE
Where's tests?

### ISSUE THREAD

**Issue by eirnym** [hyf0/ecma_regex#1]
README claims that this crate `Passes nearly 100% of the ECMAScript Test Suite tests`. Where's tests? and which exactly tests are not passing? Could you please add whole tests automation?

**Comment by hyfdev** [comment]
Well. Under the hood, `ecma_regex` is acutally using the [`Regex` engine](https://github.com/hyf0/libregexp-sys) of quickjs. So tests are done by quickjs. You could refer to https://github.com/bellard/quickjs.

For this crate, tests would be made for some edge cases or API stability. The correctness is ensured by quickjs itself.

**Comment by eirnym** [comment]
Thank you, I've read the API of this crate and the test results of QuickJS. 

I see that put all tests here would be pointless, but it would be useful to provide respective links. 

Some tests though would be explanatory how to use the API as currently there's practically no documentation at the moment. 

Also version 0.0.2 seems to be "too low to be stable" while crate you created as a library wrapper looks stable and mature enough. 

As per API, it's good for me as long as proper documentation is provided.

**Comment by hyfdev** [comment]
Yeah. Thanks for advice. The `0.0.2` version means that I haven't think trough how to maintain this repo. I'm evaluating if this crate is really useful. From myself, I found that `regex` crate is always the practical choice.

**Comment by eirnym** [comment]
I'd automate maintenance as much as I could: some tests, code formatting, enable dependabot/renovate and make releases from time to time (also semi-automated). Rust in comparison to other languages has a very useful check if a project needs to be next major/minor version.

**Comment by eirnym** [comment]
As per usability, ECMAScript regex is a part of some standards and Rust regex is not covers some of the cases, which has been made deliberately.
